sources/math.spec.move

```diff
@@ -4,6 +4,61 @@ spec rationalmath::decimal {
         ensures result.scale == d2.scale;
         ensures result.value == d1.value + d2.value;
         aborts_if d1.scale != d2.scale with ERR_DIFFERENT_SCALE;
-        aborts_if d1.value + d2.value > MAX_U128 with EXECUTION_FAILURE;
+        aborts_if d1.value + d2.value > MAX_U256 with EXECUTION_FAILURE;
     }
+
+    spec sub {
+        ensures result.scale == larger.scale;
+        ensures result.scale == smaller.scale;
+        ensures result.value == larger.value - smaller.value;
+        aborts_if larger.scale != smaller.scale with ERR_DIFFERENT_SCALE;
+        aborts_if smaller.value > larger.value with EXECUTION_FAILURE;
+    }
+
+    spec mul {
+        ensures result.scale == d1.scale;
+        ensures result.value == d1.value * d2.value;
+        aborts_if d1.value * d2.value > MAX_U256 with EXECUTION_FAILURE;
+        aborts_if spec_denominator(d2) == 0 with EXECUTION_FAILURE;
```

Comment:
by definition, spec_denominator() cannot be zero, what happens if you remove this this line from the spec?

Reply:
If we only remove this line, there is a prover error. However, adding an ensure condition that the output of denominator is always > 0 solves the issue